### PR TITLE
CDAP-9141 add a sparkprogram plugin that runs a spark program

### PR DESCRIPTION
**Pull request by albertshau**
The new plugin type behaves just like an action plugin type
except it runs a spark program instead of a custom action. The
actual spark code to run comes from the plugin. The workflow just
instantiates the plugin class and runs its main method

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RUT881

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/ExternalSparkProgram.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      if (!key.equals(PROGRAM_ARGS)) {
+        sparkConf.set(key, val);
+      }
+    }
+
+    context.setSparkConf(sparkConf);
+  }
+
+  @Override
+  public void destroy() {
```
Comment (sagarkapare) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/ExternalSparkProgram.java]: no need to override this right?

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/JavaSparkMainWrapper.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      String[] args = programArgs == null ?
+        RuntimeArguments.toPosixArray(sec.getRuntimeArguments()) : programArgs.split(" ");
+      Method mainMethod = mainClass.getMethod("main", String[].class);
+      Object[] methodArgs = new Object[1];
+      methodArgs[0] = args;
+      mainMethod.invoke(null, methodArgs);
+    }
+  }
+
+  private String getProgramArgs(JavaSparkExecutionContext sec, String stageName) {
```
Comment (sagarkapare) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/JavaSparkMainWrapper.java]: This method need to be marked with `@Nullable`

**Review by sagarkapare — COMMENTED** [review]
minor comments otherwise LGTM 👍

**Review by sagarkapare — APPROVED** [review]
LGTM 👍

**Pull request merged**